### PR TITLE
bali-phy 3.4

### PR DESCRIPTION
**Pull request by bredelings**
- [x] Have you followed the [guidelines for contributing](https://github.com/brewsci/homebrew-bio/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/brewsci/homebrew-bio/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source FORMULA`, where `FORMULA` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict FORMULA` (after doing `brew install FORMULA`)?

-----

### CODE REVIEW

**Comment by bredelings** [comment]
Hi,
Can someone help me please?

I'm trying to run `brew audit brewsci/bio/bali-phy --online` but I'm getting:

File: Formula/bali-phy.rb [hunk 1]
```diff
@@ -2,8 +2,12 @@ class BaliPhy < Formula
...
   desc "Bayesian co-estimation of phylogenies and multiple alignments"
   homepage "http://www.bali-phy.org/"
-  url "https://github.com/bredelings/BAli-Phy/archive/3.3.tar.gz"
-  sha256 "84132dd0ec8a62eb6697374c62847e4b84082cbf09977f0e157e8407db010ccc"
+  url "https://github.com/bredelings/BAli-Phy/archive/3.4.tar.gz"
+  sha256 "99ae139fe9df0e6b704f6195c62fd79207f79b2c08eb72ec5dff676ac06ee4b8"
+
+  head do
+    url "https://github.com/bredelings/BAli-Phy.git"
+  end
```
Comment (sjackman) [inline Formula/bali-phy.rb]: Try…
```ruby
  url "https://github.com/bredelings/BAli-Phy/archive/3.4.tar.gz"
  sha256 "99ae139fe9df0e6b704f6195c62fd79207f79b2c08eb72ec5dff676ac06ee4b8"
  head "https://github.com/bredelings/BAli-Phy.git"
```

**Comment by sjackman** [comment]
Try…
```sh
brew vendor-install ruby
```

File: Formula/bali-phy.rb (hunk 1 above)
Comment (bredelings) [reply Formula/bali-phy.rb]: Thanks!  This worked.

**Comment by sjackman** [comment]
```
[35/331] Compiling C++ object 'external/boost/lib/program_options/3fd6668@@boost_program_options@sta/src_parsers.cpp.o'.
FAILED: external/boost/lib/program_options/3fd6668@@boost_program_options@sta/src_parsers.cpp.o 
g++-5 -Iexternal/boost/lib/program_options/3fd6668@@boost_program_options@sta -Iexternal/boost/lib/program_options -I../external/boost/lib/program_options -fdiagnostics-color=always -pipe -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wnon-virtual-dtor -std=c++14 -O3 -DHAVE_CONFIG_H -Wall -Wextra -Wno-sign-compare -Woverloaded-virtual -Wstrict-aliasing -Wno-unknown-pragmas -Wno-maybe-uninitialized -DNDEBUG -DNDEBUG_DP -O3 -funroll-loops -ffast-math -fPIC -isystem ../external/boost/include  -MD -MQ 'external/boost/lib/program_options/3fd6668@@boost_program_options@sta/src_parsers.cpp.o' -MF 'external/boost/lib/program_options/3fd6668@@boost_program_options@sta/src_parsers.cpp.o.d' -o 'external/boost/lib/program_options/3fd6668@@boost_program_options@sta/src_parsers.cpp.o' -c ../external/boost/lib/program_options/src/parsers.cpp
{standard input}: Assembler messages:
{standard input}:3862: Warning: end of file not at end of a line; newline inserted
{standard input}:4692: Warning: ignoring changed section attributes for .text.unlikely._ZNSt6vectorIN5boost15program_options12basic_optionIwEESaIS3_EED2Ev
g++-5: internal compiler error: Killed (program cc1plus)
```
The usual cause of this error is that GCC is using more memory than CircleCI provides. Could you please try changing?
```diff
-    flags += %w[-j 4] if ENV["CIRCLECI"]
+    flags << "-j4" if ENV["CI"]
```
The CircleCI environment variable is being filtered out by new-ish environment filtering code.

File: Formula/bali-phy.rb [hunk 2]
```diff
@@ -22,7 +22,7 @@ class BaliPhy < Formula
   def install
     flags = %w[-C build install]
     # Reduce memory usage for Circle CI
-    flags += %w[-j 4] if ENV["CIRCLECI"]
+    flags += %w[-j 2] if ENV["CIRCLECI"]
```
Comment (sjackman) [inline Formula/bali-phy.rb]: You'll need to change `CIRCLECI` to `CI` for now. I've opened a PR https://github.com/Homebrew/brew/pull/5402 to address it.

Comment (sjackman) [reply Formula/bali-phy.rb]: PR https://github.com/Homebrew/brew/pull/5402 was merged.

File: Formula/bali-phy.rb [hunk 3]
```diff
@@ -1,9 +1,11 @@
 class BaliPhy < Formula
   # cite Suchard_2006: "http://dx.doi.org/10.1093/bioinformatics/btl175"
+  # cite Redelings_2014: https://dx.doi.org/10.1093%2Fmolbev%2Fmsu174
```
Comment (sjackman) [inline Formula/bali-phy.rb]: Could you please change both `%2F` to `/` ?

File: Formula/bali-phy.rb [hunk 4]
```diff
@@ -21,7 +23,7 @@ class BaliPhy < Formula
   def install
     flags = %w[-C build install]
     # Reduce memory usage for Circle CI
-    flags += %w[-j 4] if ENV["CIRCLECI"]
+    flags << "-j2" if ENV["CIRCLECI"]
```
Comment (sjackman) [inline Formula/bali-phy.rb]: `CIRCLECI` should now be `HOMEBREW_CIRCLECI`. Sorry for the back and forth.
Can you please try `-j4`? Before the PR upstream was merged, it was defaulting to `-j8`, which used too much memory. `-j4` may work.

Comment (sjackman) [reply Formula/bali-phy.rb]:
```suggestion
    flags << "-j4" if ENV["HOMEBREW_CIRCLECI"]
```

**Comment by sjackman** [comment]
Once you've made your changes, please squash your PR down to a single commit. You can use `git rebase -i origin/master && git push bredelings patch-1 --force` for this purpose. If you're using the GitHub web interface, or these instructions don't mean anything to you, don't worry about squashing. I can do it when I merge the PR.

File: Formula/bali-phy.rb (hunk 4 above)
Comment (bredelings) [reply Formula/bali-phy.rb]: Done.

File: Formula/bali-phy.rb (hunk 3 above)
Comment (bredelings) [reply Formula/bali-phy.rb]: Done.

**Comment by sjackman** [comment]
Merged. Thank you for your contributions to Brewsci/bio, Benjamin! 🏆

**Pull request closed without merging**